skyportal/models/earthquake.py

```diff
@@ -0,0 +1,183 @@
+__all__ = ['EarthquakeEvent', 'EarthquakeNotice', 'EarthquakePrediction']
+
+import sqlalchemy as sa
+from sqlalchemy.orm import relationship
+from sqlalchemy.orm import deferred
+
+from baselayer.app.models import (
+    Base,
+    AccessibleIfUserMatches,
+)
+
+from baselayer.app.env import load_env
+
+_, cfg = load_env()
+
+
+class EarthquakeNotice(Base):
+    """Earthquake notice information"""
+
+    update = delete = AccessibleIfUserMatches('sent_by')
+
+    sent_by_id = sa.Column(
+        sa.ForeignKey('users.id', ondelete='CASCADE'),
+        nullable=False,
+        index=True,
+        doc="The ID of the User who created this EarthquakeEvent.",
+    )
+
+    sent_by = relationship(
+        "User",
+        foreign_keys=sent_by_id,
+        back_populates="earthquakenotices",
+        doc="The user that saved this EarthquakeEvent",
+    )
+
+    content = deferred(
+        sa.Column(sa.LargeBinary, nullable=True, doc='Raw QuakeML content')
+    )
+
+    event_id = sa.Column(
+        sa.String,
+        sa.ForeignKey("earthquakeevents.event_id"),
+        nullable=False,
+        comment='Earthquake ID',
+    )
+
+    lat = sa.Column(sa.Float, nullable=False, comment='Latitude')
+
+    lon = sa.Column(sa.Float, nullable=False, comment='Longitude')
+
+    depth = sa.Column(
+        sa.Float,
+        nullable=False,
+        comment='Depth relative to sea level (positive values as depth increases) [m]',
+    )
+
+    magnitude = sa.Column(
+        sa.Float, nullable=False, comment='Earthquake (Moment) Magnitude', index=True
+    )
+
+    date = sa.Column(
+        sa.DateTime, nullable=False, comment='UTC event timestamp', index=True
+    )
+
+
+class EarthquakePrediction(Base):
+    """Earthquake prediction information"""
+
+    event_id = sa.Column(
+        sa.Integer,
+        sa.ForeignKey("earthquakeevents.id"),
+        nullable=False,
+        comment='Earthquake ID',
+    )
+
+    detector_id = sa.Column(
+        sa.Integer,
+        sa.ForeignKey('mmadetectors.id'),
+        nullable=False,
+        comment='Multimessenger Astronomical Detector id',
+    )
+
+    d = sa.Column(sa.Float, nullable=False, comment='Distance [km]')
+
+    p = sa.Column(sa.DateTime, nullable=False, comment='P-wave time')
+
+    s = sa.Column(sa.DateTime, nullable=False, comment='S-wave time')
+
+    r2p0 = sa.Column(sa.DateTime, nullable=False, comment='R-2.0 km/s-wave time')
+
+    r3p5 = sa.Column(sa.DateTime, nullable=False, comment='R-3.5 km/s-wave time')
+
+    r5p0 = sa.Column(sa.DateTime, nullable=False, comment='R-5.0 km/s-wave time')
+
+    rfamp = sa.Column(
+        sa.Float, nullable=False, comment='Earthquake amplitude predictions [m/s]'
+    )
+
+    lockloss = sa.Column(
+        sa.INT, nullable=False, comment='Earthquake lockloss prediction'
```

Comment:
why is this an INT? Might add some more info in the comment string.

Reply:
Yeah changed this and added more info.